Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py

```diff
@@ -0,0 +1,450 @@
+        self._setChildAlgorithmPropertyIfProvided(alg, Prop.OUTPUT_WS_BINNED)
+        self._setChildAlgorithmPropertyIfProvided(alg, Prop.OUTPUT_WS_LAM)
+        alg.setProperty("InputWorkspace", input_workspace)
+        alg.setProperty("FirstTransmissionRun", first_trans_workspace)
+        alg.setProperty("SecondTransmissionRun", second_trans_workspace)
+        alg.execute()
+        return alg
+
+    def _removePrefix(self, workspace, isTrans):
+        """Remove the TOF prefix from the given workspace name"""
+        prefix = self._transPrefix if isTrans else self._tofPrefix
+        prefix_len = len(prefix)
+        name_start = workspace[:prefix_len]
+        if len(workspace) > prefix_len and name_start == prefix:
+            return workspace[prefix_len:]
+        else:
+            return workspace
+
+    def _finalize(self, child_alg):
+        """Set our output properties from the results in the given child algorithm"""
+        self._setOutputWorkspace(Prop.OUTPUT_WS, child_alg)
+        self._setOutputWorkspace(Prop.OUTPUT_WS_BINNED, child_alg)
+        self._setOutputWorkspace(Prop.OUTPUT_WS_LAM, child_alg)
+
+    def _setOutputWorkspace(self, property_name, child_alg):
+        """Set the given output property from the result in the given child algorithm,
+        if it exists"""
+        value = child_alg.getPropertyValue(property_name)
+        if value:
+            self.setPropertyValue(property_name, value)
+            self.setProperty(property_name, child_alg.getProperty(property_name).value)
+
+
+def _setRunNumberForWorkspace(workspace_name, run_number):
+    """Set the run number in the sample log for the given workspace"""
+    AddSampleLog(Workspace=workspace_name, LogName='run_number', LogText=str(run_number),
+                 LogType='String')
+
+
+def _throwIfNotValidReflectometryEventWorkspace(workspace_name):
+    workspace=AnalysisDataService.retrieve(workspace_name)
+    if isinstance(workspace, WorkspaceGroup):
+        raise RuntimeError('Slicing workspace groups is not supported')
+    if not workspace.run().hasProperty('proton_charge'):
+        raise RuntimeError('Cannot slice workspace: run must contain proton_charge')
+
+
+def _monitorWorkspace(workspace):
+    """Return the associated monitor workspace name for the given workspace"""
+    return workspace + '_monitors'
+
+
+def _getRunNumberAsString(workspace_name):
+    """Get the run number for a workspace. If it's a workspace group, get
+    the run number from the first child workspace."""
+    try:
+        workspace = AnalysisDataService.retrieve(workspace_name)
+        if not isinstance(workspace, WorkspaceGroup):
+            return str(workspace.getRunNumber())
+        # Get first child in the group
+        return str(workspace[0].getRunNumber())
+    except:
+        raise RuntimeError('Could not find run number for workspace ' + workspace_name)
+
+
+def _isEventWorkspace(workspace_name):
```

Comment:
`_isEventWorkspace` and `isWorkspace2D` look alike.

Maybe you can replace them by `_setWorkspaceID`

```
def _setWorkspaceID(workspace_name, id):
    workspace = AnalysisDataService.retrieve(workspace_name)
    if isinstance(workspace, WorkspaceGroup):
        return workspace[0].id() == id
    else:
        return workspace.id() == id
```